internal/cmd/commands/authmethodscmd/ldap_funcs.go

```diff
@@ -0,0 +1,440 @@
+	default:
+		*opts = append(*opts, authmethods.WithLdapAuthMethodInsecureTls(false))
+	}
+
+	switch c.flagDiscoverDn {
+	case true:
+		*opts = append(*opts, authmethods.WithLdapAuthMethodDiscoverDn(true))
+	default:
+		*opts = append(*opts, authmethods.WithLdapAuthMethodDiscoverDn(false))
+	}
+
+	switch c.flagAnonGroupSearch {
+	case true:
+		*opts = append(*opts, authmethods.WithLdapAuthMethodAnonGroupSearch(true))
+	default:
+		*opts = append(*opts, authmethods.WithLdapAuthMethodAnonGroupSearch(false))
+	}
+
+	switch c.flagUpnDomain {
+	case "":
+	case "null":
+		*opts = append(*opts, authmethods.DefaultLdapAuthMethodUpnDomain())
+	default:
+		*opts = append(*opts, authmethods.WithLdapAuthMethodUpnDomain(c.flagUpnDomain))
+	}
+
+	switch c.flagUserDn {
+	case "":
+	case "null":
+		*opts = append(*opts, authmethods.DefaultLdapAuthMethodUserDn())
+	default:
+		*opts = append(*opts, authmethods.WithLdapAuthMethodUserDn(c.flagUserDn))
+	}
+
+	switch c.flagUserAttr {
+	case "":
+	case "null":
+		*opts = append(*opts, authmethods.DefaultLdapAuthMethodUserAttr())
+	default:
+		*opts = append(*opts, authmethods.WithLdapAuthMethodUserAttr(c.flagUserAttr))
+	}
+
+	switch c.flagUserFilter {
+	case "":
+	case "null":
+		*opts = append(*opts, authmethods.DefaultLdapAuthMethodUserFilter())
+	default:
+		*opts = append(*opts, authmethods.WithLdapAuthMethodUserFilter(c.flagUserFilter))
+	}
+
+	switch c.flagEnableGroups {
+	case true:
+		*opts = append(*opts, authmethods.WithLdapAuthMethodEnableGroups(true))
+	default:
+		*opts = append(*opts, authmethods.WithLdapAuthMethodEnableGroups(false))
+	}
+
+	switch c.flagGroupDn {
+	case "":
+	case "null":
+		*opts = append(*opts, authmethods.DefaultLdapAuthMethodGroupDn())
+	default:
+		*opts = append(*opts, authmethods.WithLdapAuthMethodGroupDn(c.flagGroupDn))
+	}
+
+	switch c.flagGroupAttr {
+	case "":
+	case "null":
+		*opts = append(*opts, authmethods.DefaultLdapAuthMethodGroupAttr())
+	default:
+		*opts = append(*opts, authmethods.WithLdapAuthMethodGroupAttr(c.flagGroupAttr))
+	}
+
+	switch c.flagGroupFilter {
+	case "":
+	case "null":
+		*opts = append(*opts, authmethods.DefaultLdapAuthMethodGroupFilter())
+	default:
+		*opts = append(*opts, authmethods.WithLdapAuthMethodGroupFilter(c.flagGroupFilter))
+	}
+
+	switch {
+	case len(c.flagCertificates) == 0:
+	case len(c.flagCertificates) == 1 && c.flagCertificates[0] == "null":
+		*opts = append(*opts, authmethods.DefaultLdapAuthMethodCertificates())
+	default:
+		if err := validateCerts(c.flagCertificates...); err != nil {
+			c.UI.Error(fmt.Sprintf("invalid certificate: %s", err.Error()))
+			return false
+		}
+		*opts = append(*opts, authmethods.WithLdapAuthMethodCertificates(c.flagCertificates))
+	}
+
+	switch c.flagClientCertificate {
+	case "":
+	case "null":
+		*opts = append(*opts, authmethods.DefaultLdapAuthMethodClientCertificate())
+	default:
+		*opts = append(*opts, authmethods.WithLdapAuthMethodClientCertificate(c.flagClientCertificate))
```

Comment:
Shouldn't we validate this certificate too while we can?

Reply:
ty!